packages/core/src/useForm/formContext.ts

```diff
@@ -53,13 +53,13 @@ export interface BaseFormContext<TForm extends FormObject = FormObject> {
   setFieldSubmitErrors<TPath extends Path<TForm>>(path: TPath, message: Arrayable<string>): void;
   getValidationMode(): FormValidationMode;
   getSubmitErrors: () => IssueCollection[];
-  clearErrors: (path?: string) => void;
+  clearErrors<TPath extends Path<TForm>>(path?: TPath): void;
   clearSubmitErrors: (path?: string) => void;
   getValues: () => TForm;
   setValues: (newValues: Partial<TForm>, opts?: SetValueOptions) => void;
-  revertValues: () => void;
-  revertTouched: () => void;
-  revertDirty: () => void;
+  revertValues<TPath extends Path<TForm>>(path?: TPath): void;
+  revertTouched<TPath extends Path<TForm>>(path?: TPath): void;
+  revertDirty<TPath extends Path<TForm>>(path?: TPath): void;
```

Comment:
Some of these methods use the signature `path?: string`. Is there a reason why this is. Can we just use `TPath` all around when dealing with `path`?. I updated `revertValues`, `revertTouched` and `revertDirty` to accept an optional `TPath`.

I imagine `clearSubmitErrors` can be updated as well?

Reply:
These are mostly internal functions, so strict typing them won't give us much benefits, so having relaxed signatures make things a bit easier. 

What do you think? I don't have a strong opinion here.

Reply:
I think the reason it would be good to type it as `TPath` is because we would be passing a `TPath` from `reset`. But I see, what you mean.

Reply:
Sure, as long as we don't resort to casting then let's do that.